packages/workflow/src/interfaces.ts

```diff
@@ -165,19 +165,68 @@ export interface ContinueAsNewOptions {
   searchAttributes?: SearchAttributes;
 }
 
+/**
+ * Specifies:
+ * - whether cancellation requests are sent to the Child
+ * - whether and when a {@link CanceledFailure} is thrown from {@link executeChild} or
+ *   {@link ChildWorkflowHandle.result}
+ *
+ * @default {@link ChildWorkflowCancellationType.WAIT_CANCELLATION_COMPLETED}
+ */
 export enum ChildWorkflowCancellationType {
+  /**
+   * Don't send a cancellation request to the Child. Don't throw an error.
+   */
   ABANDON = 0,
+
+  /**
+   * Send a cancellation request to the Child. Immediately throw the error.
+   */
   TRY_CANCEL = 1,
+
+  /**
+   * Send a cancellation request to the Child. Throw the error if and when the Child's cancellation is completed.
```

Comment:
I'd note that the child might not cooperate and might end up in either failed / cancelled or completed states (or never complete).